spring-kafka/src/main/java/org/springframework/kafka/support/converter/BatchMessagingMessageConverter.java

```diff
@@ -205,8 +204,8 @@ public Message<?> toMessage(List<ConsumerRecord<?, ?>> records, @Nullable Acknow
 				raws.add(record);
 			}
 		}
-		if (infos.size() > 0) {
-			rawHeaders.put(KafkaHeaders.LISTENER_INFO, infos);
+		if (info != null) {
```

Comment:
May it happen that batch is empty, but we still want to see such an info?

Reply:
The container will never submit an empty batch (unless the user has an interceptor that returns one). It could also be the result of applying a filter.

In either case, there is no way to get the info if there are no records to convert. I added a note to the docs.

Reply:
OK. I see now:
```
		private void invokeBatchListener(final ConsumerRecords<K, V> recordsArg) {
			ConsumerRecords<K, V> records = checkEarlyIntercept(recordsArg);
			if (records == null || records.count() == 0) {
				return;
			}
```

Thanks